resources-manager/src/AdaptivePruningManager.cpp

```diff
@@ -0,0 +1,76 @@
+#include "AdaptivePruningManager.hpp"
+#include "Bitmap.hpp"
+#include "ReplicaConfig.hpp"
+#include "concord.cmf.hpp"
+
+using namespace concord::performance;
+
+AdaptivePruningManager::AdaptivePruningManager(
+    const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
+    const std::chrono::duration<double, std::milli> &interval,
+    const std::shared_ptr<bftEngine::impl::IInternalBFTClient> &bftClient)
+    : repId(bftEngine::ReplicaConfig::instance().getreplicaId()),
+      resourceManager(resourceManager),
+      mode(LEGACY),
+      interval(interval),
+      bftClient(bftClient) {
+  start();
+}
+
+AdaptivePruningManager::~AdaptivePruningManager() { stop(); }
+
+void AdaptivePruningManager::notifyReplicas(const long double &rate, const uint64_t batchSize) {
+  concord::messages::ReconfigurationRequest rreq;
+
+  concord::messages::PruneTicksChangeRequest pruneRequest;
+  pruneRequest.sender_id = bftClient->getClientId();
+  // numOfConnectedReplicas requires cluster size. How can I get it?
+  // pruneRequest.tick_period_seconds = rate / bftClient->numOfConnectedReplicas();
+  pruneRequest.batch_blocks_num = batchSize;
+
+  rreq.command = pruneRequest;
+  rreq.sender = bftClient->getClientId();
+
+  std::vector<uint8_t> serialized_req;
+  rreq.signature = {};
+  concord::messages::serialize(serialized_req, rreq);
+
+  // auto sig = signer_->sign(std::string(serialized_req.begin(), serialized_req.end()));
```

Comment:
Let me know, how to obtain certificate here, cause I don't belive that it is not avialable somewhere.

Reply:
Take a look here https://github.com/vmware/concord-bft/blob/master/bftengine/src/bftengine/ReplicaImp.cpp#L5287